core/amber/src/main/scala/edu/uci/ics/texera/web/resource/auth/UserResource.scala

```diff
@@ -63,6 +105,52 @@ class UserResource {
     Response.ok().build()
   }
 
+  @POST
+  @Path("/google-login")
+  def googleLogin(@Session session: HttpSession, request: GoogleUserLoginRequest): Response = {
+    // get authorization code from request
+    var code = request.authoCode
+
+    // use authorization code to get tokens
+    try {
+      val tokenResponse = new GoogleAuthorizationCodeTokenV4Request(
+        TRANSPORT,
+        JSON_FACTORY,
+        CLIENT_ID,
+        CLIENT_SECRET,
+        code,
+        "postmessage"
+      ).execute();
+      // get the id token
+      val idToken: GoogleIdToken = tokenResponse.parseIdToken()
+      // get the payload of id token
+      val payload = idToken.getPayload
+      // get the subject of the payload, use this value as a key to identify a user.
+      val userId = payload.getSubject
+      // get the name of the user
+      val userName = payload.get("name").asInstanceOf[String]
+
+      // store Google user id in database if it does not exist
+      if (this.googleUserDao.fetchOneByUid(userId) == null) {
+        val googleUser = new GoogleUser(userId, userName)
+        this.googleUserDao.insert(googleUser)
+      } else if (this.googleUserDao.fetchOneByUid(userId).getName != userName) {
+        this.googleUserDao.fetchOneByUid(userId).setName(userName)
+      }
+
+      // get access token and refresh token (used foe access Google API Service)
+      val access_token = tokenResponse.getAccessToken
+      val refresh_token = tokenResponse.getRefreshToken
+
```

Comment:
are those two variables being used anywhere?

Reply:
no, i just commented them out for later reference.